### ISSUE TITLE
comparisons some questions around images

### ISSUE THREAD

**Issue by cdstg** [ansible-collections/community.docker#316]
I was hoping for some help on this in regards to the comparisons i see if i change config the container recreates.

But I m confused about the image itself if that changes  in the registry how is the comparison made ?
Or is it not possible.

Eg  lets say I have MariaDB 10.4  container running and there is and update yes the tag may become 10.4.20 and my running one is maybe 10.4.18

But there is also a tag 10.4  on that new one in the registry.

Basically i want to run a job every so often to check for updates in the registry  and then recreate the container if needed for that major version

I was wondering if the image compare checks the hash of the current image to the new one or its simply just by tag number

hopefully i made myself clear

**Comment by felixfontein** [comment]
The docker_container module checks the image ID (hash) that the container is using against the local image that was last pulled, assuming an image name is given to the module. (Here, name includes the tag, which is by default `latest` if you do not specify one.) If the hash does not match, it will recreate the container. When you provide an image name to the module, you can ask the module to pull the image before checking (`pull: true`). In that case it will first try to pull the image of that name from the registry, and then check whether the current image of that name in the local Docker daemon (which should now be the latest from the registry for that name) matches the one (by hash) that the current container is using.

The module does not allow you to check whether new tags are available on the registry side, you have to do that differently. There's no module or plugin in this collection that will do that for you, since the Docker daemon offers no functionality for that. You have to talk directly to the registry, for example with the `uri` module (https://docs.ansible.com/ansible/latest/collections/ansible/builtin/uri_module.html). If something follows the Docker registry v2 format, you can do something like `curl https://registry-domain/v2/namespace/repository/tags/list`. This doesn't seem to be available for Docker Hub (`index.docker.io`) anymore, it seems to want authentication for this operation :shrug:. For other registries such as quay.io it works though: `curl https://quay.io/v2/bedrock/ubuntu/tags/list` will give you a JSON for all `quay.io/bedrock/ubuntu` tags.

**Comment by cdstg** [comment]
thanks this is perfect just the pull: true  is what i had missing

**Comment by felixfontein** [comment]
Great! In that case I'll close this issue.
